### PR TITLE
Add encoding format to routing-endpoint iothub command

### PR DESCRIPTION
**Pull request by anusapan**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [*] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [*] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/command_modules/azure-cli-iot/HISTORY.rst [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 Release History
 ===============
+0.3.6
++++++
+* Add encoding format to routing-endpoint command.
```
[the remark this answers is not in the document]
Comment (tjprescott) [reply src/command_modules/azure-cli-iot/HISTORY.rst]: Yes

File: src/command_modules/azure-cli-iot/setup.py [hunk 2]
```diff
@@ -14,7 +14,7 @@
     logger.warn("Wheel is not available, disabling bdist_wheel hook")
     cmdclass = {}
 
-VERSION = "0.3.5"
+VERSION = "0.3.6"
```
[the remark this answers is not in the document]
Comment (anusapan) [reply src/command_modules/azure-cli-iot/setup.py]: fixed

File: src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/_params.py [hunk 3]
```diff
@@ -134,6 +134,9 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
                    help='Name of the storage container.')
         c.argument('endpoint_type', arg_type=get_enum_type(EndpointType),
                    options_list=['--endpoint-type', '--type', '-t'], help='Type of the Routing Endpoint.')
+        c.argument('encoding', options_list=['--encoding', '--en'], arg_type=get_enum_type(EncodingFormat),
+                   help='Encoding format for the container. You may select JSON or AVRO. The default is AVRO. '
```
[the remark this answers is not in the document]
Comment (anusapan) [reply src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/_params.py]: resolved

File: src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/_params.py [hunk 4]
```diff
@@ -134,6 +134,9 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
                    help='Name of the storage container.')
         c.argument('endpoint_type', arg_type=get_enum_type(EndpointType),
                    options_list=['--endpoint-type', '--type', '-t'], help='Type of the Routing Endpoint.')
+        c.argument('encoding', options_list=['--encoding', '--en'], arg_type=get_enum_type(EncodingFormat),
```
[the remark this answers is not in the document]
Comment (anusapan) [reply src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/_params.py]: fixed

File: src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/shared.py [hunk 5]
```diff
@@ -32,3 +32,12 @@ class RouteSourceType(Enum):
...
     DeviceJobLifecycleEvents = 'devicejoblifecycleevents'
+
+
+# pylint: disable=too-few-public-methods
+class EncodingFormat(Enum):
+    """
+    Type of the encoding format for the container.
+    """
+    JSON = 'json'
+    AVRO = 'avro'
```
Comment (tjprescott) [inline src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/shared.py]: Why is this here?  Isn't it simply pulled from your SDK?
